src/dune_rules/jsoo/jsoo_rules.ml

```diff
@@ -50,84 +50,122 @@ module Config : sig
+    | "double-translation" -> Some Cps
+    | _ -> None
+  ;;
+
   let of_string x =
     match x with
     | "default" -> default
     | _ ->
       List.fold_left (String.split ~on:'+' x) ~init:default ~f:(fun acc name ->
-        match String.drop_prefix ~prefix:"!" name with
-        | Some name -> set acc name false
-        | None -> set acc name true)
+        match ( String.drop_prefix ~prefix:"!" name
+              , String.drop_prefix ~prefix:"effects=" name ) with
+        | Some name, _ -> set acc name `False
+        | None, None -> set acc name `True
+        | None, Some backend ->
+            (match effects_of_string backend with
+            | Some backend -> set acc name (`Effects backend)
+            | None -> acc)
+        )
   ;;
 
   let of_flags l =
     let rec loop acc = function
       | [] -> acc
-      | "--enable" :: name :: rest -> loop (set acc name true) rest
+      | "--enable" :: name :: rest -> loop (set acc name `True) rest
       | maybe_enable :: rest when String.is_prefix maybe_enable ~prefix:"--enable=" ->
         (match String.drop_prefix maybe_enable ~prefix:"--enable=" with
-         | Some name -> loop (set acc name true) rest
+         | Some name -> loop (set acc name `True) rest
          | _ -> assert false)
-      | "--disable" :: name :: rest -> loop (set acc name false) rest
+      | "--disable" :: name :: rest -> loop (set acc name `False) rest
       | maybe_disable :: rest when String.is_prefix maybe_disable ~prefix:"--disable=" ->
         (match String.drop_prefix maybe_disable ~prefix:"--disable=" with
-         | Some name -> loop (set acc name false) rest
+         | Some name -> loop (set acc name `False) rest
          | _ -> assert false)
-      | "--toplevel" :: rest -> loop (set acc "toplevel" true) rest
+      | "--toplevel" :: rest -> loop (set acc "toplevel" `True) rest
+      | "--effects" :: "cps" :: rest -> loop (set acc "effects" `Cps) rest
+      | "--effects" :: "double-translation" :: rest ->
+          loop (set acc "effects" `Double_translation) rest
+      | maybe_effects :: rest when String.is_prefix maybe_effects ~prefix:"--effects=" ->
+          let backend =
+            Option.bind
+              (String.drop_prefix maybe_effects ~prefix:"--effects=")
+              ~f:effects_of_string
+          in
+          (match backend with
+           | Some backend -> set acc "effects" (`Effects backend)
+           | None -> loop acc rest)
       | _ :: rest -> loop acc rest
     in
     loop default l
   ;;
 
   let to_flags t =
-    List.concat_map (get t) ~f:(function
-      | "toplevel", true -> [ "--toplevel" ]
-      | "toplevel", false -> []
-      | name, true -> [ "--enable"; name ]
-      | name, false -> [ "--disable"; name ])
-  ;;
+    List.filter_opt
+      [ (match t.toplevel with Some true -> Some "--toplevel" | _ -> None)
+      ; (match t.effects with
+         | Some backend -> Some ("--effects=" ^ string_of_effects backend) | None -> None)
```

Comment:
I think we need to remember if effect was set with `--enable effects` and use the same flag here otherwise we loose backward compatibility.

Reply:
I would select the flags based on jsoo version:
```ocaml
let to_flag ~recent =
  [...]
       | Some backend ->
           Some (if recent then "--effects=" ^ string_of_effects backend else "--enable=effects")
       | None -> None)
  [...]
```

Reply:
Done.